src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonKeyValueAdapter.cs

```diff
@@ -42,7 +44,13 @@ internal class JsonKeyValueAdapter : IKeyValueAdapter
             }
             catch (JsonException)
             {
-                // If the value is not a valid JSON, treat it like regular string value
+                if (ThrowOnInvalidJson)
+                {
+                    throw new FormatException($"Invalid JSON value for key '{setting.Key}' with content type '{setting.ContentType}'. Please ensure the value is properly formatted JSON.");
+                }
+
+                logger.LogWarning($"Invalid JSON value for key '{setting.Key}' with content type '{setting.ContentType}'. Treated as a string value.");
```

Comment:
Maybe we can mention the new API in this warning log?

Reply:
Why do we need to mention the API?

Reply:
I don't think we have to add it, at the time I was thinking about the scenario where someone gets this exception for multiple key-values and how it could be hard to discover the API to disable it, but I think the reaction to seeing this message shouldn't be just to ignore it and disable strict parsing anyway.